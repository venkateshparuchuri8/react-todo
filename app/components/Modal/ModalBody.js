import styled from 'styled-components';

const ModalBody = styled.div`
  position: relative;
  -webkit-box-flex: 1;
  flex: 1 1 auto;
  padding: 1rem;
`;

export default ModalBody;
