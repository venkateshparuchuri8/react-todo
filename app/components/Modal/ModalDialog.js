import styled from 'styled-components';

const ModalDialog = styled.div`
  position: relative;
  width: auto;
  margin: .5rem auto;
  margin-top: 3rem;
  pointer-events: none;
  transition: transform .3s ease-out,-webkit-transform .3s ease-out;
  transform: translate(0,0);
  box-shadow: 0 24px 38px 3px rgba(0,0,0,0.14), 0 9px 46px 8px rgba(0,0,0,0.12), 0 11px 15px -7px rgba(0,0,0,0.2);
  max-width: ${({ size }) => {
    if (size === 'sm') {
      return '500px';
    } if (size === 'md') {
      return '700px';
    } if (size === 'login') {
      return '420px';
    } if (size === 'lg') {
      return '900px';
    } if (size === 'xl') {
      return '100%';
    }
    return '500px';
  }}
`;

export default ModalDialog;
