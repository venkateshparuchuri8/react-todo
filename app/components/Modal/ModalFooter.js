import styled from 'styled-components';

const ModalFooter = styled.div`
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 1rem;
`;

export default ModalFooter;
