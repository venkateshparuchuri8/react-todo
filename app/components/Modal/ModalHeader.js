import styled from 'styled-components';

const ModalHeader = styled.div`
  display: flex;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1rem;
  h5 {
    font-size: 30px;
    display: inline-block;
    margin: 0;
    text-transform: capitalize;
  }
  button {
    cursor: pointer;
    background-color: transparent;
    border: 0;
    padding: .7em;
    transform: rotate(45deg);
    width: 30px;
    height: 30px;
    margin: 2px;
    display: inline-block;
    vertical-align: middle;
    position: relative;
    font-style: normal;
    color: #000000;
    text-align: left;
    text-indent: -9999px;
    direction: ltr;
    transition: all .2s;
    right: 0;
    & :focus {
      outline: none;
    }
    ::before {
      content: '';
      width: 20px;
      height: 2px;
      box-shadow: inset 0 0 0 32px;
      transform: translate(-50%,-50%);
      position: absolute;
      left: 50%;
      top: 50%;
    }
    ::after {
      content: '';
      height: 20px;
      width: 2px;
      box-shadow: inset 0 0 0 32px;
      position: absolute;
      left: 50%;
      top: 50%;
      transform: translate(-50%,-50%);
    }
  }
`;

export default ModalHeader;
