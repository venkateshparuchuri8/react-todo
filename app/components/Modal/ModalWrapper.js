import styled from 'styled-components';

const ModalWrapper = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 9999;
  overflow: hidden;
  outline: 0;
  overflow-x: hidden;
  overflow-y: auto;
  padding-right: ${({ show }) => show ? '15px' : '0px'};
  padding-left: ${({ show }) => show ? '15px' : '0px'};
  background-color: rgba(0, 0, 0, .50);
  opacity: ${({ show }) => show ? 1 : 0};
  transition: opacity 0.2s ease-in;
  height: ${({ show }) => show ? 'auto' : 0};
`;

export default ModalWrapper;
