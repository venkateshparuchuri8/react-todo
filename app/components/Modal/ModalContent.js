import styled from 'styled-components';
import ModalBody from './ModalBody';
import ModalHeader from './ModalHeader';
import ModalFooter from './ModalFooter';
import Input from '../Input';
import Button from '../Button';

const ModalContent = styled.div`
  position: relative;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  flex-direction: column;
  width: 100%;
  pointer-events: auto;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid rgba(0, 0, 0, 0.2);
  outline: 0;
  &.default {
    background: #fff;
    ${ModalHeader} {
      padding: 0px;
      h5 {
        position: absolute;
        left: 50%;
        transform: translateX(-50%);
        // width: 100%;
        top: 35px;
        font-size: 19px;
      }
      button {
        position: absolute;
        z-index: 9;
        right: 15px;
        top: 15px;
        border-radius: 50%;
        color: #5d5d5d;
      }
    }
    ${ModalBody} {
      .formElement {
        padding: 30px 30px 0px 30px;
        margin-top: 40px;
        ${Input} {
          width: 100%;
          margin-bottom: 20px;
        }
        textarea {
          width: 100%;
          height: 75px;
        }
      }
      .common {
        padding: 10px;
        text-align: center;
        margin: 0;
        form {
          margin-top: 20px;
        }
      }
    }
    ${ModalFooter} {
      padding: 0px 30px 30px 30px;
      button {
        background: ${props => props.theme.primaryColor};
        color: #fff;
      }
    }
  }
  &.videoPopup {
    ${ModalBody} {
      padding: 0px;
      background: transparent;
      iframe {
        min-height: 500px;
      }
    }
    ${ModalHeader} {
      padding: 0px;
      h5 {
        position: absolute;
        left: 50%;
        transform: translateX(-50%);
        // width: 100%;
        top: 35px;
        font-size: 19px;
      }
      button {
        position: absolute;
        z-index: 9;
        right: -15px;
        top: -15px;
        border-radius: 50%;
        color: #5d5d5d;
        background: #fff;
      }
    }
  }
  &.without-header {
    background: #fff;
    ${ModalHeader} {
      padding: 0px;
      h5 {
        display: none;
      }
      button {
        position: absolute;
        z-index: 9;
        right: 15px;
        top: 15px;
        border-radius: 50%;
        // background:#fff;
        color: #5d5d5d;
      }
    }
    ${ModalBody} {
    }
    ${ModalFooter} {
    }
  }
`;

export default ModalContent;
