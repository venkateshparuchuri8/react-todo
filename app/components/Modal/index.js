import React from 'react';
import PropTypes from 'prop-types';

import ModalWrapper from './ModalWrapper';
import ModalDialog from './ModalDialog';
import ModalContent from './ModalContent';
import ModalBody from './ModalBody';
import ModalHeader from './ModalHeader';
import ModalFooter from './ModalFooter';

class Modal extends React.Component { // eslint-disable-line react/prefer-stateless-function
  renderHeader(header, showClose, onClose) {
    if (header || showClose) {
      return (
        <ModalHeader>
          {typeof header === 'string' ?
            <h5>{header}</h5>
            :
            header
          }
          <button onClick={onClose} />
        </ModalHeader>
      );
    }
    return '';
  }

  renderFooter(footer) {
    if (footer) {
      return (
        <ModalFooter>
          {footer}
        </ModalFooter>
      );
    }
    return '';
  }

  render() {
    const { show, header, children, showClose, footer, size, onClose, classname } = this.props;
    return (
      <ModalWrapper
        show={show}
      >
        <ModalDialog
          size={size}
        >
          <ModalContent
            className={classname}
          >
            {this.renderHeader(header, showClose, onClose)}
            <ModalBody>
              {children}
            </ModalBody>
            {this.renderFooter(footer)}
          </ModalContent>
        </ModalDialog>
      </ModalWrapper>
    );
  }
}

Modal.defaultProps = {
  show: false,
  header: '',
  showClose: true,
  size: 'md',
  classname: '',
  onClose: () => '',
};

Modal.propTypes = {
  show: PropTypes.bool,
  header: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
  ]),
  children: PropTypes.node,
  showClose: PropTypes.bool,
  footer: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
  ]),
  size: PropTypes.oneOf([
    'sm', 'md', 'lg', 'xl', 'login', 'null',
  ]),
  onClose: PropTypes.func,
  classname: PropTypes.string,
};

export default Modal;
