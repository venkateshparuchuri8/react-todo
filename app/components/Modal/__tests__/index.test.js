/**
 * @author vishwadeep.kapoor
 * @Date: 2018-12-27 13:17:36
 * @Last Modified by: vishwadeep
 * @Last Modified time: 2018-12-27 13:29:37
 */
/* eslint-disable no-undef */
import React from 'react';
import Modal from '../index';

describe('Modal rendering', () => {
  it('is renders correctly', () => {
    const props = { header: 'modalHeader', footer: 'modalFooter' };
    const tree = renderWithTheme(<Modal {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('is renders correctly with header', () => {
    const props = { header: <div>header</div> };
    const tree = renderWithTheme(<Modal {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('is renders correctly without header size sm', () => {
    const props = { showClose: false, size: 'sm' };
    const tree = renderWithTheme(<Modal {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('is renders correctly size login', () => {
    const props = { showClose: false, size: 'login' };
    const tree = renderWithTheme(<Modal {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('is renders correctly size lg', () => {
    const props = { showClose: false, size: 'lg' };
    const tree = renderWithTheme(<Modal {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('is renders correctly size xl', () => {
    const props = { showClose: false, size: 'xl' };
    const tree = renderWithTheme(<Modal {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('is renders correctly size null', () => {
    const props = { showClose: false, size: 'null' };
    const tree = renderWithTheme(<Modal {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
