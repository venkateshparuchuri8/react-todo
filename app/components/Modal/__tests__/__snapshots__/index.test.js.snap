// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal rendering is renders correctly 1`] = `
<div
  className="sc-bdVaJa kxpWPG"
>
  <div
    className="sc-bwzfXH hNVVMj"
    size="md"
  >
    <div
      className="sc-gzVnrw brJaDo"
    >
      <div
        className="sc-bxivhb kimRTT"
      >
        <h5>
          modalHeader
        </h5>
        <button
          onClick={[Function]}
        />
      </div>
      <div
        className="sc-htpNat cdmvmB"
      />
      <div
        className="sc-ifAKCX kOZAnx"
      >
        modalFooter
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal rendering is renders correctly size lg 1`] = `
<div
  className="sc-bdVaJa kxpWPG"
>
  <div
    className="sc-bwzfXH czYgep"
    size="lg"
  >
    <div
      className="sc-gzVnrw brJaDo"
    >
      
      <div
        className="sc-htpNat cdmvmB"
      />
      
    </div>
  </div>
</div>
`;

exports[`Modal rendering is renders correctly size login 1`] = `
<div
  className="sc-bdVaJa kxpWPG"
>
  <div
    className="sc-bwzfXH gRXUpo"
    size="login"
  >
    <div
      className="sc-gzVnrw brJaDo"
    >
      
      <div
        className="sc-htpNat cdmvmB"
      />
      
    </div>
  </div>
</div>
`;

exports[`Modal rendering is renders correctly size null 1`] = `
<div
  className="sc-bdVaJa kxpWPG"
>
  <div
    className="sc-bwzfXH gpkSsi"
    size="null"
  >
    <div
      className="sc-gzVnrw brJaDo"
    >
      
      <div
        className="sc-htpNat cdmvmB"
      />
      
    </div>
  </div>
</div>
`;

exports[`Modal rendering is renders correctly size xl 1`] = `
<div
  className="sc-bdVaJa kxpWPG"
>
  <div
    className="sc-bwzfXH dmZbNq"
    size="xl"
  >
    <div
      className="sc-gzVnrw brJaDo"
    >
      
      <div
        className="sc-htpNat cdmvmB"
      />
      
    </div>
  </div>
</div>
`;

exports[`Modal rendering is renders correctly with header 1`] = `
<div
  className="sc-bdVaJa kxpWPG"
>
  <div
    className="sc-bwzfXH hNVVMj"
    size="md"
  >
    <div
      className="sc-gzVnrw brJaDo"
    >
      <div
        className="sc-bxivhb kimRTT"
      >
        <div>
          header
        </div>
        <button
          onClick={[Function]}
        />
      </div>
      <div
        className="sc-htpNat cdmvmB"
      />
      
    </div>
  </div>
</div>
`;

exports[`Modal rendering is renders correctly without header size sm 1`] = `
<div
  className="sc-bdVaJa kxpWPG"
>
  <div
    className="sc-bwzfXH gpkSsi"
    size="sm"
  >
    <div
      className="sc-gzVnrw brJaDo"
    >
      
      <div
        className="sc-htpNat cdmvmB"
      />
      
    </div>
  </div>
</div>
`;
